hal-sim/hal_impl/mode_helpers.py

```diff
@@ -5,114 +5,114 @@
 from . import data
 hal_data = data.hal_data
 from . import functions as fns
-    
+
 def notify_new_ds_data():
     '''Called when driver station data is modified'''
     data.hooks.notifyDSData()
 
 def set_autonomous(enabled):
     '''Only designed to be called on transition'''
-    
+
     hal_data['control'].update({
         'autonomous': True,
         'test': False,
         'enabled': enabled,
         'dsAttached': True
     })
-    
+
     if enabled:
-        hal_data['time']['match_start'] = fns.hooks.getFPGATime()
+        hal_data['time']['remaining'] = 135000000
     else:
-        hal_data['time']['match_start'] = None
-    
+        hal_data['time']['remaining'] = None
+
     notify_new_ds_data()
-    
+
 
 def set_test_mode(enabled):
     '''Only designed to be called on transition'''
-    
+
     hal_data['control'].update({
         'autonomous': False,
         'test': True,
         'enabled': enabled,
         'dsAttached': True
     })
-    
-    hal_data['time']['match_start'] = None
+
+    hal_data['time']['remaining'] = None
     notify_new_ds_data()
-    
+
 
 def set_teleop_mode(enabled):
     '''Only designed to be called on transition'''
-    
+
     hal_data['control'].update({
         'autonomous': False,
         'test': False,
         'enabled': enabled,
         'dsAttached': True
     })
-    
+
     if enabled:
-        hal_data['time']['match_start'] = fns.hooks.getFPGATime() - 15000000
+        hal_data['time']['remaining'] = hal_data['time']['remaining'] + 12000000
```

Comment:
missing a 0. And if I enable autonomous (remaining = 135) then enable teleop, remaining now equals 255. Add to getFPGATime instead